src/openforms/formio/utils.py

```diff
@@ -94,3 +94,104 @@ def mimetype_allowed(mime_type: str, allowed_mime_types: List[str]) -> bool:
         return True
 
     return mime_type in allowed_mime_types
+
+
+# See https://help.form.io/userguide/forms/form-components#input-mask for the
+# semantics, and FormioUtils.getInputMask for the implementation.
+def conform_to_mask(value: str, mask: str) -> str:
+    """
+    Given an input and a Formio mask, try to conform the value to the mask.
+
+    If the value is not compatible with the mask, a ``ValueError`` is raised.
+    """
+    # loop through all the characters of the value and test them against the mask. Note
+    # that the value and mask may have different lengths due to non-input characters
+    # such as spaces/dashes... Formio only recognizes alphanumeric inputs as variable
+    # user input, the rest is part of the mask.
+    # NOTE: we don't check for numeric masks or not, let that be handled by Formio itself
+
+    # the final result characters, build from the value and mask
+    result: List[str] = []
+    char_index, mask_index = 0, 0
+    LOOP_GUARD, iterations = 1000, 0
+
+    # maps the formio mask characters to rules to test for alphanumeric chars etc.
+    TEST_RULES = {
+        "9": lambda c: c.isnumeric(),
+        "A": lambda c: c.isalpha(),
+        "a": lambda c: c.isalpha(),
```

Comment:
I notice that in `getInputMask` the `a` refers to only lower case, while `A` to both lower/upper case. Is it intentional that here `A` and `a` are the same?

Edit: I see the conversion to lowercase is done later (L145)